### DISCUSSION TITLE
Questions about commercial use of Avalonia

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#11842] [Q&A] [answered]
I am going to use Avalonia in a commercial project. And recently I read this article:
https://github.com/robloo/PublicDocs/blob/master/XAMLFrameworkComparison.md

In this article there is such a passage:
`Avalonia UI was originally entirely MIT licensed and achieved three stars. However, with the [re-licensing of the composition renderer](https://raw.githubusercontent.com/AvaloniaUI/Avalonia/master/src/Avalonia.Base/Rendering/Composition/License.md) to prohibit modification and distribution in anything other than original binary form this has lowered the score. The composition renderer is the only renderer supported in Avalonia version 11+, others were removed. This makes it prohibitive to modify Avalonia and distribute it in your own application. The team has clarified the license will "revert to MIT with v11 going GA" and this section will be updated when that happens.`

In https://raw.githubusercontent.com/AvaloniaUI/Avalonia/master/src/Avalonia.Base/Rendering/Composition/License.md , it said:
`This code may be used free of charge by any application that consumes Avalonia binary packages as a direct or indirect dependency.`

In my opinion, this sentence means that if Avalonia is used, it can only be compiled into a binary release through AOT, otherwise it cannot be used for free.

However, in Avalonia Github page, it is still MIT licence:
![image](https://github.com/AvaloniaUI/Avalonia/assets/30720036/7f5b7d0b-0dbc-4a53-897d-2194616a45ae)

Can any official Avalonia staff tell me clearly whether Avalonia UI (not XPF) is free for commercial use, or is there a charge under certain circumstances? We need to evaluate the various costs that the project will encounter when using Avalonia.

**Comment by rabbitism** [comment]
please read this discussion. 
https://github.com/AvaloniaUI/Avalonia/discussions/11182#discussioncomment-5771173

**Reply by kyuranger** [reply]
Even if the author's description of that article is wrong, I am still worried because Avalonia has never made a formal statement. After all, https://raw.githubusercontent.com/AvaloniaUI/Avalonia/master/src/Avalonia.Base/Rendering/Composition/License.md has always existed.

**Comment by timunie** [comment]
> In my opinion, this sentence means that if Avalonia is used, it can only be compiled into a binary release through AOT, otherwise it cannot be used for free.

That's not true.

@kyuranger compiling the App into AOT is not needed to be used. Only requitement is that you don't use the compositor outside an Avalonia-App.

**Comment by maxkatz6** [comment, marked as answer]
> Can any official Avalonia staff tell me clearly whether Avalonia UI (not XPF) is free for commercial use

Yes.

> is there a charge under certain circumstances

No.

As [/Composition/License.md](https://raw.githubusercontent.com/AvaloniaUI/Avalonia/master/src/Avalonia.Base/Rendering/Composition/License.md) states, you need explicit permission only if you DON'T use Avalonia binary packages as a direct or indirect dependency. Or in other words, if you use this part of the code (specifically Composition subfolder) outside of Avalonia packages.

> In my opinion, this sentence means that if Avalonia is used, it can only be compiled into a binary release through AOT, otherwise it cannot be used for free.

There is no mention of AOT. By "binary packages" it also includes MSIL binaries, i.e. dlls.
